const mongoose = require('mongoose');
const Person = mongoose.model(process.env.PERSON_MODEL);
const vulnerabilityController = require('./volunerabilityController');
const _helpers = require('../helpers/refactoringHelpers');


const getPerson = function(personId) {
    console.log("Get Person Controller called");
    const response = {
        status : process.env.HTTP_STATUS_OK,
        message: {}
    }
    if(mongoose.isValidObjectId(personId)) {
        console.log("Valid user Id");
        Person.findById(personId).exec(function(err, person) {
            if(err) {
                console.log("Error reading Person");
                response.status = process.env.HTTP_STATUS_INTERNAL_ERROR;
                response.message = err;    
            }
            if(person) {
                console.log("Found Person");
                response.status = process.env.HTTP_STATUS_OK;
                response.message = person.id; 
            }
            else {
                console.log("Person is null");
                response.status = process.env.process.env.HTTP_STATUS_NOTFOUND;
                response.message = {"message" : "Person with given Id not found"}
            }
            console.log(response.message);
            return response;
            });

    }
    else {
        response.status = process.env.HTTP_STATUS_NOTFOUND;
        response.message = {"message" : "Invalid user Id"}
        return response;
    }

}

const addPerson = async function(req, res) {
    const {email, userName, firstName, lastName, role, password} = req.body;
    const response = _helpers.initResponse();
    const person = new Person ({
        email : email,
        userName: userName,
        firstName : firstName,
        lastName: lastName,
        role : role,
        password : password
    }) ;
    try {
        
        const savePerson = await person.save();
        response.status = process.env.HTTP_STATUS_OK;
        response.message = savePerson;
        res.status(response.status).json(response.message);


    } catch (error) {
        response.status = process.env.HTTP_STATUS_INTERNAL_ERROR;
        response.message = error
        res.status(response.status).json(response.message);
    }
}

module.exports = {
    getPerson : getPerson,
    addPerson:addPerson
}